### ISSUE TITLE
[iOS] Show spell check toolbar when caret touches misspelled word

### ISSUE THREAD

**Issue by camsim99** [flutter/flutter#119538]
On iOS, a spell check toolbar containing suggestions for misspelled words should show when the caret moves to touch a misspelled word (and only if it is misspelled). Thus, to follow the behavior of iOS, the Cupertino style spell check suggestions toolbar should show when the caret moves to touch a word, and disappear when the caret moves away.

Fixing this depends on https://github.com/flutter/flutter/issues/119537.

**Comment by jmagman** [comment]
@camsim99 do you have a screenshot or recording of what this should look like? It's a little hard to follow the textual description. 🙂

**Comment by camsim99** [comment]
An example:

https://user-images.githubusercontent.com/43054281/218532526-01c4f6eb-8a5f-42a9-9b43-c74b6c459e41.MOV

**Comment by jmagman** [comment]
> to follow the behavior of iOS, the Cupertino style spell check suggestions toolbar should show when the caret moves to touch a word, and disappear when the caret moves away

Shouldn't it work the same way on iOS no matter if Material or Cupertino is used (meaning make it match by platform, not widget style)?

**Comment by camsim99** [comment]
> Shouldn't it work the same way on iOS no matter if Material or Cupertino is used (meaning make it match by platform, not widget style)?

Oops! Yes I misspoke. I'll modify the issue.
